src/components/swap/TradePrice.tsx

```diff
@@ -41,7 +42,9 @@ export default function TradePrice({ price, showInverted, setShowInverted }: Tra
 
   let formattedPrice: string
   try {
-    formattedPrice = showInverted ? price.toSignificant(4) : price.invert()?.toSignificant(4)
+    formattedPrice = showInverted
+      ? formatTransactionAmount(parseFloat(price.toSignificant(4)))
```

Comment:
are we losing precision here?

Reply:
good point, aded new fn priceToPreciseFloat